olive/workflows/run/config.py

```diff
@@ -240,12 +234,6 @@ def validate_data_configs_with_hf_model(cls, v, values):
     def validate_evaluators(cls, v, values):
         for idx, metric in enumerate(v.get("metrics", [])):
             v["metrics"][idx] = _resolve_data_config(metric, values, "data_config")
-
-            data_dir_config = v["metrics"][idx].get("user_config", {}).get("data_dir", None)
-            if isinstance(data_dir_config, dict):
-                if _have_aml_client(data_dir_config, values):
-                    data_dir_config["config"]["azureml_client"] = values["azureml_client"]
-                v["metrics"][idx]["user_config"]["data_dir"] = data_dir_config
```

Comment:
This might still be required if `eval_func_args` are to contain the `data_dir` & `batch_size` as discussed in the earlier PR.

Reply:
The root validator works directly on the whole workflow config. It applies the insertion recursively so that case is automatically covered.